.gitignore

```diff
@@ -1,7 +1,22 @@
 *.o
 *.a
 *~
+*.so
 led-image-viewer
 demo
 minimal-example
 text-example
+
+# visual studio
+*.suo
+*.sln
+*.csproj
+[Dd]ebug/
+[Rr]elease/
+x64/
+x86/
+[Bb]in/
+[Oo]bj/
+[Ll]og/
+.vs/
+AssemblyInfo.cs
```

Comment:
yes, these folders and files are generated by visual studio. I added them to the gitignore to make it easier for people using VS to contribute. They can be removed from the gitignore if desired.